pkg/cmd/server/admin/decrypt.go

```diff
@@ -0,0 +1,158 @@
+package admin
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/spf13/cobra"
+
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	pemutil "github.com/openshift/origin/pkg/cmd/util/pem"
+)
+
+const DecryptCommandName = "decrypt"
+
+type DecryptOptions struct {
+	Data    *pem.Block
```

Comment:
As an API for re-use, it seems more useful to take something like `ExternalizableString` type for congruence and this command can simply choose which bits it wants to fill in.

Asking for a `*pem.Block` doesn't seem like something that most callers are going to have sitting around and caring about.

I could also see a case for forcing them to pass you `[]byte`